### PR TITLE
[sw/silicon_creator] Add flash_ctrl_creator_info_pages_lockdown()

### PR DESCRIPTION
**Pull request by alphan**
This PR adds `flash_ctrl_creator_info_pages_lockdown()` to disable access to silicon creator info pages until next reset. The pages that are locked down are those that should not be accessible after ROM_EXT (taken from the flash info map spreadsheet). Please note that `kFlashCtrlInfoPageCreatorSecret` is locked down in `flash_ctrl_init()`.

Unit test changes rely on the preprocessor and auto-generated register offset defines to generate a map for tests. I think we can do something similar in `flash_ctrl.c` to generate a switch to be able to use hardened info page enums (instead of relying on low hamming weight bit fields and arithmetic operations). Please let me know what you think.

### CODE REVIEW

File: sw/device/silicon_creator/lib/drivers/flash_ctrl_unittest.cc [hunk 1]
```diff
@@ -17,18 +17,94 @@
...
+            bank_,                                                                 \
+            page_,                                                                 \
+            FLASH_CTRL_BANK##bank_##_INFO0_PAGE_CFG_SHADOWED_##page_##_REG_OFFSET, \
+            FLASH_CTRL_BANK##bank_##_INFO0_REGWEN_##page_##_REG_OFFSET,            \
+        },                                                                         \
+  }
+
+  static const std::map<flash_ctrl_info_page_t, InfoPage> *const kInfoPages =
+      []() {
+        auto info_pages = new std::map<flash_ctrl_info_page_t, InfoPage>{
```
Comment (mcy) [inline sw/device/silicon_creator/lib/drivers/flash_ctrl_unittest.cc:49]: I don't think you need this lambda. You can just do `= new` directly.

File: sw/device/silicon_creator/lib/drivers/flash_ctrl_unittest.cc [hunk 2]
```diff
@@ -17,18 +17,94 @@
...
+
 class FlashCtrlTest : public mask_rom_test::MaskRomTest {
  protected:
   uint32_t base_ = TOP_EARLGREY_FLASH_CTRL_CORE_BASE_ADDR;
   mask_rom_test::MockAbsMmio mmio_;
   mask_rom_test::MockSecMmio sec_mmio_;
 };
 
+class InfoPagesTest : public FlashCtrlTest {};
+TEST_F(InfoPagesTest, NumberOfPages) { EXPECT_EQ(InfoPages().size(), 20); }
```
Comment (mcy) [inline sw/device/silicon_creator/lib/drivers/flash_ctrl_unittest.cc:64]: EXPECT_THAT(... SizeIs(20));

File: sw/device/silicon_creator/lib/drivers/flash_ctrl_unittest.cc [hunk 3]
```diff
@@ -17,18 +17,94 @@
...
+
+TEST_F(InfoPagesTest, PagesPerBank) {
+  std::array<uint32_t, 2> pages_per_bank = {0, 0};
+  for (auto const &it : InfoPages()) {
```
Comment (mcy) [inline sw/device/silicon_creator/lib/drivers/flash_ctrl_unittest.cc:68]: const auto& =P

File: sw/device/silicon_creator/lib/drivers/flash_ctrl_unittest.cc [hunk 4]
```diff
@@ -17,18 +17,94 @@
...
+    const uint32_t bank = it.second.bank;
+    EXPECT_EQ(bank, static_cast<uint32_t>(bitfield_bit32_read(
+                        it.first, FLASH_CTRL_INFO_PAGE_BIT_BANK)));
+    EXPECT_LE(bank, 1);
+    ++pages_per_bank[bank];
+  }
+
+  EXPECT_EQ(pages_per_bank[0], 10);
```
Comment (mcy) [inline sw/device/silicon_creator/lib/drivers/flash_ctrl_unittest.cc:76]: EXPECT_THAT(pages_per_bank, Each(10))

**Review by mcy — APPROVED** [review]
LG modulo test expectation nits.

**Review by moidx — APPROVED** [review]

**Pull request merged**